libminifi/include/FlowController.h

```diff
@@ -107,42 +109,39 @@ class FlowController : public core::controller::ControllerServiceProvider, publi
     return initialized_.load();
   }
   // Start to run the Flow Controller which internally start the root process group and all its children
-  virtual int16_t start();
-  virtual int16_t pause() {
+  int16_t start() override;
+  int16_t pause() override {
     return -1;
   }
   // Unload the current flow YAML, clean the root process group and all its children
-  virtual int16_t stop(bool force, uint64_t timeToWait = 0);
-  virtual int16_t applyUpdate(const std::string &source, const std::string &configuration);
-  virtual int16_t drainRepositories() {
-
+  int16_t stop(bool force, uint64_t timeToWait = 0) override;
+  int16_t applyUpdate(const std::string &source, const std::string &configuration) override;
+  int16_t drainRepositories() override {
     return -1;
   }
 
-  virtual std::vector<std::shared_ptr<state::StateController>> getComponents(const std::string &name);
+  std::vector<std::shared_ptr<state::StateController>> getComponents(const std::string &name) override;
 
-  virtual std::vector<std::shared_ptr<state::StateController>> getAllComponents();
+  std::vector<std::shared_ptr<state::StateController>> getAllComponents() override;
 
-  virtual int16_t clearConnection(const std::string &connection);
+  int16_t clearConnection(const std::string &connection) override;
 
-  virtual int16_t applyUpdate(const std::string &source, const std::shared_ptr<state::Update> &updateController) {
-    return -1;
-  }
+  int16_t applyUpdate(const std::string &source, const std::shared_ptr<state::Update>&) override { return -1; }
   // Asynchronous function trigger unloading and wait for a period of time
-  virtual void waitUnload(const uint64_t timeToWaitMs);
+  virtual void waitUnload(uint64_t timeToWaitMs);
   // Unload the current flow xml, clean the root process group and all its children
   virtual void unload();
   // Load new xml
   virtual void reload(std::string yamlFile);
   // update property value
   void updatePropertyValue(std::string processorName, std::string propertyName, std::string propertyValue) {
     if (root_ != nullptr)
-      root_->updatePropertyValue(processorName, propertyName, propertyValue);
+      root_->updatePropertyValue(std::move(processorName), std::move(propertyName), std::move(propertyValue));
```

Comment:
(i) I would defer the string copy as long as possible, especially when they are passed on unmodified

Reply:
so rather than move, const ref parameters

Reply:
I don't want to extend the scope of this PR to anything outside of this file. Currently `ProcessGroup::updatePropertyValue` takes the strings by value.

It would be nice if they went through the whole call graph as a const ref, but they are taken by value here and I don't want to change 2-3 more classes to fix this unrelated issue in this PR. A move makes more sense locally IMO.

Given the above, if you still want me to change this function to take its parameters by const ref, in anticipation of further future changes, I'm fine with doing so.